### ISSUE TITLE
Failed pipeline keeps being shown as running

### ISSUE THREAD

**Issue by JulianGro** [woodpecker-ci/woodpecker#1061]
### Component

web-ui

### Describe the bug

When a pipeline fails, it properly reports to fail to GitHub, but the Woodpecker WebUI still shows the pipeline as running.

### System Info

```shell
{"source":"https://github.com/woodpecker-ci/woodpecker","version":"next-37146058"}

My agents are running the same version.


# docker-compose.yml
version: '3'

services:
  woodpecker-server:
    image: woodpeckerci/woodpecker-server:next
    ports:
      - 80:80
      - 443:443
      - 9000:9000
    volumes:
      - woodpecker-server-data:/var/lib/woodpecker/
    environment:
      - WOODPECKER_OPEN=true
      - WOODPECKER_HOST=https://ci.moto9000.moe
      - WOODPECKER_LETS_ENCRYPT=true
      - WOODPECKER_GITHUB=true
      - WOODPECKER_GITHUB_CLIENT=REDACTED
      - WOODPECKER_GITHUB_SECRET=REDACTED
      - WOODPECKER_AGENT_SECRET=REDACTED
      - WOODPECKER_ADMIN=REDACTED
      - WOODPECKER_REPO_OWNERS=REDACTED,REDACTED

volumes:
  woodpecker-server-data:
```

### Additional context

```shell
`ci.moto9000.moe` has an AAAA record, but IPv6 is not configured on Docker.

For some reason I cannot upload anything:
https://data.moto9000.moe/m/2022-07-31_23.30.25_ci.moto9000.moe_d0c217bc1720.png
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

**Comment by JulianGro** [comment]
> Even if you reload the page?

Yes

**Comment by JulianGro** [comment]
`build_status` in `builds` is `running`.
The `build_error` column is empty on all builds as well. (Even though no build has ever succeeded on this instance.)
`build_status` says `killed` on all other builds. I assume this specifies that I killed them and that the status for failing is different.

**Comment by JulianGro** [comment]
For reference, this is using the standard settings (SQLite inside Docker)

**Comment by JulianGro** [comment]
It has already occured over 4 times. For me it happens every time I run a build and it fails.

**Comment by JulianGro** [comment]
Well I do see a lot of errors but nothing that seems related.
I will turn on debug logging and trigger the issue again.

**Comment by JulianGro** [comment]
Here you go: https://paste.ubuntu.com/p/rrhJZVhtJH/
If you don't see anything in there, I guess I can make it fail sooner and change the loglevel to trace.

**Comment by JulianGro** [comment]
Hmm. I actually tried running with trace, but it doesn't find the config file in my repository anymore. Even manually specifying it doesn't work anymore. I tried going back to debug log level again but still nothing.

Any chance that the next branch got broken in the last three days?

**Comment by 6543** [comment]
this is a posible glitch where if you have a running agent and just kill it, the agent can never report back that the step did fail and so it wont get updated ...

hotfix for UI -> hit cancle button, now it's cancled

long term fix is the "agent heartbeat" and similar functions ...

**Comment by 6543** [comment]
might be fixed at #3151

**Comment by 6543** [comment]
the other mentioned issue is now tracked at:  #3157
